src/cmds/restic/cmd_find.go

```diff
@@ -84,7 +85,93 @@ func parseTime(str string) (time.Time, error) {
 	return time.Time{}, errors.Fatalf("unable to parse time: %q", str)
 }
 
-func findInTree(repo *repository.Repository, pat findPattern, id restic.ID, prefix string, snapshotID *string) error {
+type statefulOutput struct {
+	ListLong bool
+	JSON     bool
+	inuse    bool
+	newsn    *restic.Snapshot
+	oldsn    *restic.Snapshot
+	hits     int
+}
+
+func (s *statefulOutput) PrintJSON(prefix string, node *restic.Node) {
+	if node.Type != "file" && node.Type != "dir" {
+		// How to represent links and devices in json?
```

Comment:
Add a `type` field to the JSON output and populate it with the type, so the caller can decide whether or not they are interested in other things as files.